[quant] Onnx export of quantized models with new API

supriyar opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#38736 [quant] Onnx export of quantized models with new API**

Summary:
qconv2d and qlinear APIs were changed recently so updating the scale code accordingly

Test Plan:
python test/onnx/test_pytorch_onnx_caffe2_quantized.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21647724](https://our.internmc.facebook.com/intern/diff/D21647724)

supriyar committed:
[quant] Onnx export of quantized models with new API
Summary:
qconv2d and qlinear APIs were changed recently so updating the scale code accordingly

Test Plan:
python test/onnx/test_pytorch_onnx_caffe2_quantized.py

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

supriyar committed:
Update on "[quant] Onnx export of quantized models with new API"


Summary:
qconv2d and qlinear APIs were changed recently so updating the scale code accordingly

Test Plan:
python test/onnx/test_pytorch_onnx_caffe2_quantized.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21647724](https://our.internmc.facebook.com/intern/diff/D21647724)

[ghstack-poisoned]

vkuzo commented on torch/csrc/jit/passes/onnx/unpack_quantized_weights.cpp:
```diff
@@ -60,27 +62,25 @@ double getScaleFromInput(Node* input_node) {
     TORCH_CHECK(
-        input_node->inputs().size() > 7,
+        input_node->inputs().size() > 2,
         "quantized::conv2d expected scale to be 8th input");
```
nit: does this string need to change too?

vkuzo commented on torch/csrc/jit/passes/onnx/unpack_quantized_weights.cpp:
```diff
@@ -60,27 +62,25 @@ double getScaleFromInput(Node* input_node) {
     TORCH_CHECK(
-        input_node->inputs().size() > 6,
+        input_node->inputs().size() > 2,
         "quantized::conv2d_relu expected scale to be 7th input");
```
and this one

supriyar committed:
Update on "[quant] Onnx export of quantized models with new API"


Summary:
qconv2d and qlinear APIs were changed recently so updating the scale code accordingly

Test Plan:
python test/onnx/test_pytorch_onnx_caffe2_quantized.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21647724](https://our.internmc.facebook.com/intern/diff/D21647724)

[ghstack-poisoned]

xchani commented:
Thanks for your effort. This pr solves my problem, but I need to revert line 53 `scale = toIValue(input_node->inputs()[2]);` back to `scale = toIValue(input_node->inputs()[3]);`, otherwise will encounter error descributed as below.
```
Traceback (most recent call last):
    ...
    File "/opt/conda/lib/python3.7/site-packages/torch/onnx/utils.py", line 171, in _optimize_graph
      torch._C._jit_pass_onnx_unpack_quantized_weights(graph, params_dict)
RuntimeError: bad optional access
```
Hope you can fix this in the future.
